src/runtime_src/driver/common/config_reader.cpp

```diff
@@ -54,27 +54,52 @@ get_self_path()
 #endif
 }
 
+/*
+ * Look for xrt.ini and i not found look for legacy sdaccel.ini
+ */
 static std::string
-get_ini_path()
+verify_ini_path(const boost::filesystem::path& dir)
 {
   try {
-    auto ini_path = boost::filesystem::path(value_or_empty(std::getenv("SDACCEL_INI_PATH")));
-    // Support SDACCEL_INI_PATH with/without actual filename
-    if (ini_path.filename() != "sdaccel.ini")
-      ini_path /= "sdaccel.ini";
-    if (boost::filesystem::exists(ini_path))
-      return ini_path.string();
-    auto exe_path = boost::filesystem::path(get_self_path()).parent_path()/"sdaccel.ini";
-    if (boost::filesystem::exists(exe_path))
-      return exe_path.string();
-    auto self_path = boost::filesystem::current_path()/"sdaccel.ini";
-      if (boost::filesystem::exists(self_path))
-        return self_path.string();
-  }  catch (const boost::filesystem::filesystem_error& e){ }
+    auto file_path = dir / "xrt.ini";
+    if (boost::filesystem::exists(file_path))
+      return file_path.string();
 
+    file_path = dir / "sdaccel.ini";
+    if (boost::filesystem::exists(file_path))
+      return file_path.string();
+  }
+  catch (const boost::filesystem::filesystem_error& e) {
+  }
   return "";
 }
 
+static std::string
+get_ini_path()
+{
+  std::string full_path;
+  try {
+    auto ini_path = boost::filesystem::path(value_or_empty(std::getenv("XRT_INI_PATH")));
+    full_path = verify_ini_path(ini_path);
```

Comment:
this seems to assume XRT_INI_PATH contains a directory only. But the original code seems to indicate that it may contain path to the ini file. Intentional behavior change here?

Reply:
yes, I do think you should add the check from old line 63/64.   I know I have scripts that specify full path to ini file, which seems more intuitive that just specifying the directory with the ini file.

Reply:
If ENV is set (XRT or SDACCEL) we now require the file to be present or else we give up. No fall back to other directories any more. Note this change is currently limited to config_reader in XRT. I still need to update OCL version.